### PR TITLE
Change command line parsing to the Qt way

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

File: mscore/musescore.cpp [hunk 1]
```diff
@@ -4530,8 +4474,14 @@ int main(int argc, char* av[])
...
       QCoreApplication::setOrganizationDomain("musescore.org");
-      QCoreApplication::setApplicationName("MuseScoreDevelopment");
-      QCoreApplication::setApplicationVersion(VERSION);
+      if (MuseScore::unstable()) {
+            QCoreApplication::setApplicationName("MuseScoreDevelopment");
+            QCoreApplication::setApplicationVersion(QString("Music Score Editor\nUnstable Prerelease for Version %1; Build %2").arg(VERSION).arg(revision));
+            }
+      else {
+            QCoreApplication::setApplicationName("MuseScore2");
+            QCoreApplication::setApplicationVersion(QString("Music Score Editor; Version %1; Build %2").arg(VERSION).arg(revision));
```
Comment (lasconic) [inline mscore/musescore.cpp]: That will not work. ApplicationVersion needs to be VERSION.

**Comment by Jojo-Schmitz** [comment]
Mind to explain why this is needed?
We could do a long version --long-version (and idea for a short switch?) via the old printVersion() function and a short version, -v, --version, via the new QCommandLineParser::addVersionOption(), how does that sound?

File: mscore/musescore.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [inline mscore/musescore.cpp]: Mind to explain why this is needed?
 We could do a long version --long-version (and idea for a short switch?) via the old printVersion() function and a short version, -v, --version, via the new QCommandLineParser::addVersionOption(), how does that sound?

**Comment by lasconic** [comment]
It changes the behavior. setApplicationVersion was holding version before that. This Qt property can be used in many ways, for example by libraries for auto update. Also Qt could use it in configuration file or whatever. I wouldn't change it just to have a nice usage(). I would prefer to have a less good usage() then.

**Comment by Jojo-Schmitz** [comment]
Well, without that change behaviors changes too, of the -v output.
Anyway, here it goes...

**Pull request merged**